src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,139 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final double LOAD_FACTOR = 0.75;
+    private int size;
+    private int index;
+    private Node<K, V>[] storage;
+    private Node<K, V> temp;
+
+    public MyHashMap() {
+        storage = new Node[DEFAULT_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        checkLoadedCapacity();
+        if (key == null) {
+            putByIndex(null, value, 0);
+        } else {
+            index = definePosition(key);
+            putByIndex(key, value, index);
+        }
     }
 
     @Override
     public V getValue(K key) {
+        if (size > 0) {
+            index = definePosition(key);
+            temp = storage[index];
+            while (temp != null) {
+                if (Objects.equals(key, temp.key)) {
+                    return temp.value;
+                }
+                temp = temp.next;
+            }
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    public V remove(K key) {
+        index = definePosition(key);
+        temp = storage[index];
+        V result = null;
+        if (Objects.equals(storage[index].key, key)) {
```

Comment:
The `remove` method assumes that `storage[index]` is not null, which can lead to a `NullPointerException` if the key is not found. Consider adding a null check for `storage[index]` before accessing its `key`.